Add ability to set the playback mode

christarazi opened the pull request.

ambientsound commented:
Hi @christarazi, thank you for the PR.

I think it is convenient to support setting these parameters through options, I would like to see this functionality in PMS. However, there are a few issues I would like you to address before merging:

* There must be a null check on `CurrentMpdClient`, otherwise the client will crash when there is no MPD connection.
* The options should be updated when the server changes them, and they should always be in sync with the server settings.

christarazi commented:
> There must be a null check on CurrentMpdClient, otherwise the client will crash when there is no MPD connection.

Fixed in 69fc22d. Adding the nil check at the top (before the switch) caused PMS to not load correctly on start up, for some reason. So I resorted to putting the retrieving and checking in each case statement. Please let me know if you have another way in mind.

> The options should be updated when the server changes them, and they should always be in sync with the server settings.

Fixed in a2204b5

kimtore commented:
The design of this feature irks me, because I don't think the options should be so tightly coupled with MPD.
Closing this five years after the fact. :man_shrugging:
